plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java

```diff
@@ -294,4 +295,29 @@ public void testSelectInformationSchemaColumns()
         // Isolate this test to avoid problem described in https://github.com/trinodb/trino/issues/10846
         executeExclusively(super::testSelectInformationSchemaColumns);
     }
+
+    @Override
+    @Test
+    @Disabled
+    public void testComplexJoinPushdown()
+    {
+        // Join pushdown in SQL server produces different behavior depending
+        // on whether the plan contains an inner join node with a filter vs
+        // the filter on top of the join.
+        //
+        // The connector is producing column stats inconsistently, based on timing.
+        // This causes join reordering to sometimes fire before join pushdown happens.
+        // When it fires, it produces a slightly different plan that the connector
+        // accepts for pushdown, which causes the assertions to fail.
+
+        // TODO: fix join pushdown in SQL server connector by making filter(<f>, innerjoin(l, r, true)) work the same as innerjoin(l, r, <f>)
+    }
+
+    @Override
+    @Test
+    public void testJoinPushdown()
+    {
+        // TODO: fix join pushdown in SQL server connector
+        // See testComplexJoinPushdown() for explanation
```

Comment:
https://github.com/trinodb/trino/issues/21621

Reply:
Redshift has the same issue: https://github.com/trinodb/trino/actions/runs/8749315702/job/24010872826
TestRedshiftConnectorTest didn't fail because it is disabled on CI.